bolt/lib/Core/DIEBuilder.cpp

```diff
@@ -535,11 +523,12 @@ void DIEBuilder::finish() {
     finalizeCU(*CU, TypeUnitStartOffset);
   }
 
+  uint64_t UnitSizes = UnitSize;
```

Comment:
Let's just separate this into two variables.
One for computing CU offsets here, and another one for DebugNamesTable.
I think this kind of fragile to rely on updateDebugNamesTable to update UnitSize and to make sure for monolithic case it's invoked right after finish()